extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/lib/structure/devices/deviceParameters/constants/SensorEnum.java

```diff
@@ -0,0 +1,244 @@
+     * | 19 | Wind direction | degrees | 0 | 511,875 | 0,54 |
+     * | 20 | Precipitation | Milliliter per square meter (ml/m2) | 0 | 102,375 | 0,025 |
+     * | 21 | Carbon Dioxide | Parts per million (ppm) | 1 | 131446,795 | logarithmic: ppm = 10 * (x/800), x = 800 * log
+     * (ppm) |
+     * | 25 | Sound pressure level | Decibel (dB) | 0 | 255,938 | 0,25/4 |
+     * | 50 | Room temperature set point | Kelvin (K) | 230 | 332,375 | 0,025 | (not correct)
+     * | 51 | Room temperature control variable | Percent (%) | 0 | 102,375 | 0,025 |
+     * | 64 | Output current (H) | Ampere (mA) | 0 | 16380 | 4 |
+     * | 65 | Power consumption | Volt-Ampere (VA) | 0 | 4095 | 1 |
+     *
+     * Note: Point 20 precipitation, mm/m2 doesn't exist as unit, so it's liter or milliliter per square meter (ml/m2 or
+     * l/m2)
+     * otherwise it's only mm what is the same as l/m2)
+     *
+     */
+    ACTIVE_POWER((short) 4, "watt", "W", 0, 4095, 1, "%d"),
+    OUTPUT_CURRENT((short) 5, "ampere", "mA", 0, 4095, 1, "%d"),
+    ELECTRIC_METER((short) 6, "kilowatt_hours", "kWh", 0, (float) 40.95, (float) 0.01, "%.3f"),
+    TEMPERATURE_INDOORS((short) 9, "kelvin", "K", 230, (float) 332.375, (float) 0.25, "%.2f"),
+    TEMPERATURE_OUTDOORS((short) 10, "kelvin", "K", 230, (float) 332.375, (float) 0.25, "%.2f"),
+    BRIGHTNESS_INDOORS((short) 11, "lux", "Lx", 1, (float) 131446.795, 800, "%.3f"),
+    BRIGHTNESS_OUTDOORS((short) 12, "lux", "Lx", 1, (float) 131446.795, 800, "%.3f"),
+    RELATIVE_HUMIDITY_INDOORS((short) 13, "percent", "%", 0, (float) 102.375, (float) 0.025, "%.2f"),
+    RELATIVE_HUMIDITY_OUTDOORS((short) 14, "percent", "%", 0, (float) 102.375, (float) 0.025, "%.2f"),
+    AIR_PRESSURE((short) 15, "pascal", "hPa", 0, (float) 1223.75, (float) 0.25, "%.2f"),
+    WIND_SPEED((short) 18, "meters_per_second", "m/s", 0, (float) 102.375, (float) 0.025, "%.2f"),
+    WIND_DIRECTION((short) 19, "degrees", "°", 0, (float) 511.875, (float) 0.54, "%.2f"),
+    PRECIPITATION((short) 20, "millimeter_per_square_meter", "mm/m2", 0, (float) 102.375, (float) 0.025, "%.3f"),
+    CARBON_DIOXIDE((short) 21, "parts_per_million", "ppm", 1, (float) 131446.795, 800, "%.3f"),
+    SOUND_PRESSURE_LEVEL((short) 25, "decibel", "dB", 0, (float) 255.938, (float) 0.0625, "%.2f"),
+    ROOM_TEMPERATURE_SET_POINT((short) 50, "kelvin", "K", 230, (float) 332.375, (float) 0.025, "%.2f"),
+    ROOM_TEMPERATURE_CONTROL_VARIABLE((short) 51, "kelvin", "K", 230, (float) 332.375, (float) 0.25, "%.2f"),
+    OUTPUT_CURRENT_H((short) 64, "ampere", "mA", 0, 16380, 4, "%d"),
+    POWER_CONSUMPTION((short) 65, "volt_ampere", "VA", 0, 4095, 1, "%d");
+
+    private final Short sensorType;
+    private final String unit;
+    private final String unitShortcut;
+    private final String pattern;
+    private final Integer min;
+    private final Float max;
+    private final Float resolution;
+
+    static final HashMap<Short, SensorEnum> SENSOR_ENUMS = new HashMap<Short, SensorEnum>();
```

Comment:
`static final Map<Short, SensorEnum> SENSOR_ENUMS = new HashMap<Short, SensorEnum>();`

Please also have a look at other places where you define lists or maps so you use the interface type in the definition and the specific one in the initialization.